### PR TITLE
docs: update buildpack support

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- add recent tools, which are now compatible
- marked as draft, until all open PR's are merged

## Context
- #13683
- #16976
- #16986
- #16989
- #16990
- #16991
- #16996

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -149,8 +149,11 @@ Supported tools for dynamic install are:
 - `jb`
 - `jsonnet-bundler`
 - `mix`
-- `npm`
+- `npm` (including `yarn`, `pnpm` and `lerna`)
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:152]:
```suggestion
- `npm`
- `yarn`
- `pnpm`
- `lerna`
```

How about this? This way the list always has one item per line?

Comment (viceice) [reply docs/usage/self-hosted-configuration.md:152]: i think we should add also node and python then

Comment (HonkingGoose) [reply docs/usage/self-hosted-configuration.md:152]: Sounds good to me. I'll let you add those to the list.

You'll probably need to sort the list from A-Z after adding the new items. 🙂

**Review by rarkins — DISMISSED** [review]

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -148,9 +148,17 @@ Supported tools for dynamic install are:
...
 - `mix`
+- `node`
 - `npm`
 - `pipenv`
+- `pip-compile`
+- `pip_requirements`
+- `pnpm`
+- `poetry`
+- `python`
+- `yarn`
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:161]: This list is not sorted from A-Z. 😉 

To fix:

1. Open the file in VSCode.
2. Select the list items.
3. Use the <kbd>F1</kbd> key to open the Command Palette.
4. Search for `Sort lines ascending`.
5. Select "Sort lines ascending" to fix the list.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**